demosplan/DemosPlanCoreBundle/Repository/ProcedureRepository.php

```diff
@@ -820,38 +829,34 @@ public function generateObjectValues($procedure, array $data)
     }
 
     /**
-     * @param int  $exactlyDaysToGo number of day, in which the procedures ends
+     * @param int  $exactlyDaysToGo number of days, in which the procedures ends
      * @param bool $internal        check for institution phases. false checks public phases
      *
      * @return Procedure[] containing the procedures, which are ending in the given number of days
      *
      * @throws Exception
      */
-    public function getListOfSoonEnding(int $exactlyDaysToGo, $internal = true): array
+    public function getListOfSoonEnding(int $exactlyDaysToGo, bool $internal = true): array
     {
-        if (!is_numeric($exactlyDaysToGo)) {
-            $this->getLogger()->warning('getListIfSoonEnding needs an integer as parameter. The given parameter: '.$exactlyDaysToGo.' is not numeric.');
-            throw new Exception('getListIfSoonEnding needs an integer as parameter. The given parameter: '.$exactlyDaysToGo.' is not numeric.');
-        }
+        $resultProcedureList = [];
 
         try {
+            $phase = $internal ? 'phase' : 'publicParticipationPhase';
+
+            $query = $this->createFluentQuery();
+            $query->getConditionDefinition()
+                ->propertyHasValue(false, ['deleted'])
+                ->propertyHasValue(false, ['master'])
+                ->propertyHasValue(false, ['masterTemplate'])
+                ->propertyHasValueAfterNow([$phase, 'endDate']);
+
+            $notEndedProcedures = $query->getEntities();
```

Comment:
you loose the order

Reply:
good catch
Yes, I was aware of it, but the order isn't necessary. Anyway, to keep the logical change small I restore the order:
79a24514f9731e3898e1126390e55d721fb98346